make:entity --regenerate and classes using traits

I'm using traits in my entities like blameable, timestampable etc. I have getters/setters in those traits, so I don't have to define them in every entity. The command adds the getters/setters to my entities, although I have them already in the traits, and even adds the private property definitions again... I have tried to rewrite everything to exactly match the generated code, but the generator still inserts the functions again, so my guess is that it ignores the trait.

Maybe there is some other solution, and using traits is not the way to move forward, or this is just a bug. I need some advice or confirmation in both cases :) Thanks!

@weaverryan , I am having some struggle with running test for this bundle. Specifically, running just `EntityRegeneratorTest` leads to class resolution error:

```
vendor/bin/simple-phpunit tests/Doctrine/EntityRegeneratorTest.php 
PHPUnit 6.5.8 by Sebastian Bergmann and contributors.

Testing Symfony\Bundle\MakerBundle\Tests\Doctrine\EntityRegeneratorTest
..E                                                                 3 / 3 (100%)

Time: 6.13 seconds, Memory: 26.00MB

There was 1 error:

1) Symfony\Bundle\MakerBundle\Tests\Doctrine\EntityRegeneratorTest::testXmlRegeneration
Doctrine\Common\Persistence\Mapping\MappingException: Class 'Symfony\Bundle\MakerBundle\Tests\tmp\current_project\src\Entity\XOther' does not exist

/Users/andrewtch/Documents/Work/maker-bundle/vendor/doctrine/common/lib/Doctrine/Common/Persistence/Mapping/MappingException.php:96
/Users/andrewtch/Documents/Work/maker-bundle/vendor/doctrine/common/lib/Doctrine/Common/Persistence/Mapping/RuntimeReflectionService.php:41
/Users/andrewtch/Documents/Work/maker-bundle/vendor/doctrine/common/lib/Doctrine/Common/Persistence/Mapping/AbstractClassMetadataFactory.php:282
/Users/andrewtch/Documents/Work/maker-bundle/vendor/doctrine/common/lib/Doctrine/Common/Persistence/Mapping/AbstractClassMetadataFactory.php:312
/Users/andrewtch/Documents/Work/maker-bundle/vendor/doctrine/orm/lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php:78
/Users/andrewtch/Documents/Work/maker-bundle/vendor/doctrine/common/lib/Doctrine/Common/Persistence/Mapping/AbstractClassMetadataFactory.php:217
/Users/andrewtch/Documents/Work/maker-bundle/vendor/doctrine/common/lib/Doctrine/Common/Persistence/Mapping/AbstractClassMetadataFactory.php:115
/Users/andrewtch/Documents/Work/maker-bundle/src/Doctrine/DoctrineHelper.php:124
/Users/andrewtch/Documents/Work/maker-bundle/src/Doctrine/EntityRegenerator.php:45
/Users/andrewtch/Documents/Work/maker-bundle/tests/Doctrine/EntityRegeneratorTest.php:99
/Users/andrewtch/Documents/Work/maker-bundle/tests/Doctrine/EntityRegeneratorTest.php:63

ERRORS!
Tests: 3, Assertions: 16, Errors: 1.
```

Main test suite (which runs like forever) also returns errors:

```
vendor/bin/simple-phpunit
PHPUnit 6.5.8 by Sebastian Bergmann and contributors.

Testing Project Test Suite
..........E....................EE..E.....................EEEEEE  63 / 175 ( 36%)
EEEEEEEEEEEE................................................... 126 / 175 ( 72%)
.................................................               175 / 175 (100%)

Time: 6.4 minutes, Memory: 36.00MB

There were 22 errors:

1) Symfony\Bundle\MakerBundle\Tests\Doctrine\EntityRegeneratorTest::testXmlRegeneration
Doctrine\Common\Persistence\Mapping\MappingException: Class 'Symfony\Bundle\MakerBundle\Tests\tmp\current_project\src\Entity\XOther' does not exist

/Users/andrewtch/Documents/Work/maker-bundle/vendor/doctrine/common/lib/Doctrine/Common/Persistence/Mapping/MappingException.php:96
/Users/andrewtch/Documents/Work/maker-bundle/vendor/doctrine/common/lib/Doctrine/Common/Persistence/Mapping/RuntimeReflectionService.php:41
/Users/andrewtch/Documents/Work/maker-bundle/vendor/doctrine/common/lib/Doctrine/Common/Persistence/Mapping/AbstractClassMetadataFactory.php:282
/Users/andrewtch/Documents/Work/maker-bundle/vendor/doctrine/common/lib/Doctrine/Common/Persistence/Mapping/AbstractClassMetadataFactory.php:312
/Users/andrewtch/Documents/Work/maker-bundle/vendor/doctrine/orm/lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php:78
/Users/andrewtch/Documents/Work/maker-bundle/vendor/doctrine/common/lib/Doctrine/Common/Persistence/Mapping/AbstractClassMetadataFactory.php:217
/Users/andrewtch/Documents/Work/maker-bundle/vendor/doctrine/common/lib/Doctrine/Common/Persistence/Mapping/AbstractClassMetadataFactory.php:115
/Users/andrewtch/Documents/Work/maker-bundle/src/Doctrine/DoctrineHelper.php:124
/Users/andrewtch/Documents/Work/maker-bundle/src/Doctrine/EntityRegenerator.php:45
/Users/andrewtch/Documents/Work/maker-bundle/tests/Doctrine/EntityRegeneratorTest.php:99
/Users/andrewtch/Documents/Work/maker-bundle/tests/Doctrine/EntityRegeneratorTest.php:63

2) Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest::testCommands with data set "command" (Symfony\Bundle\MakerBundle\Test\MakerTestDetails Object (...))
file_get_contents(/Users/andrewtch/Documents/Work/maker-bundle/tests/tmp/cache/maker_f0ddcd02a3/maker_f0ddcd02a3.json): failed to open stream: No such file or directory

/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:235
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestCase.php:63
/Users/andrewtch/Documents/Work/maker-bundle/tests/Maker/FunctionalTest.php:51

Caused by
Exception: Error running command: "php bin/console cache:clear --no-ansi". Output: "
Fatal error: Class 'Symfony\Component\Console\Application' not found in /Users/andrewtch/Documents/Work/maker-bundle/tests/tmp/cache/maker_f0ddcd02a3/vendor/symfony/framework-bundle/Console/Application.php on line 30

Call Stack:
    0.0007     401392   1. {main}() /Users/andrewtch/Documents/Work/maker-bundle/tests/tmp/cache/maker_f0ddcd02a3/bin/console:0
    0.0039     491816   2. class_exists() /Users/andrewtch/Documents/Work/maker-bundle/tests/tmp/cache/maker_f0ddcd02a3/bin/console:14
    0.0040     491896   3. spl_autoload_call() /Users/andrewtch/Documents/Work/maker-bundle/tests/tmp/cache/maker_f0ddcd02a3/bin/console:14
    0.0040     491976   4. Composer\Autoload\ClassLoader->loadClass() /Users/andrewtch/Documents/Work/maker-bundle/tests/tmp/cache/maker_f0ddcd02a3/bin/console:14
    0.0042     492168   5. Composer\Autoload\includeFile() /Users/andrewtch/Documents/Work/maker-bundle/tests/tmp/cache/maker_f0ddcd02a3/vendor/composer/ClassLoader.php:322
    0.0054     522624   6. include('/Users/andrewtch/Documents/Work/maker-bundle/tests/tmp/cache/maker_f0ddcd02a3/vendor/symfony/framework-bundle/Console/Application.php') /Users/andrewtch/Documents/Work/maker-bundle/tests/tmp/cache/maker_f0ddcd02a3/vendor/composer/ClassLoader.php:444

". Error: "PHP Fatal error:  Class 'Symfony\Component\Console\Application' not found in /Users/andrewtch/Documents/Work/maker-bundle/tests/tmp/cache/maker_f0ddcd02a3/vendor/symfony/framework-bundle/Console/Application.php on line 30
PHP Stack trace:
PHP   1. {main}() /Users/andrewtch/Documents/Work/maker-bundle/tests/tmp/cache/maker_f0ddcd02a3/bin/console:0
PHP   2. class_exists() /Users/andrewtch/Documents/Work/maker-bundle/tests/tmp/cache/maker_f0ddcd02a3/bin/console:14
PHP   3. spl_autoload_call() /Users/andrewtch/Documents/Work/maker-bundle/tests/tmp/cache/maker_f0ddcd02a3/bin/console:14
PHP   4. Composer\Autoload\ClassLoader->loadClass() /Users/andrewtch/Documents/Work/maker-bundle/tests/tmp/cache/maker_f0ddcd02a3/bin/console:14
PHP   5. Composer\Autoload\includeFile() /Users/andrewtch/Documents/Work/maker-bundle/tests/tmp/cache/maker_f0ddcd02a3/vendor/composer/ClassLoader.php:322
PHP   6. include() /Users/andrewtch/Documents/Work/maker-bundle/tests/tmp/cache/maker_f0ddcd02a3/vendor/composer/ClassLoader.php:444
" in /Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestProcess.php:55
Stack trace:
#0 /Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php(139): Symfony\Bundle\MakerBundle\Test\MakerTestProcess->run()
#1 /Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestCase.php(31): Symfony\Bundle\MakerBundle\Test\MakerTestEnvironment->runMaker()
#2 /Users/andrewtch/Documents/Work/maker-bundle/tests/Maker/FunctionalTest.php(51): Symfony\Bundle\MakerBundle\Test\MakerTestCase->executeMakerCommand(Object(Symfony\Bundle\MakerBundle\Test\MakerTestDetails))
#3 [internal function]: Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest->testCommands(Object(Symfony\Bundle\MakerBundle\Test\MakerTestDetails))
#4 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/Framework/TestCase.php(1072): ReflectionMethod->invokeArgs(Object(Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest), Array)
#5 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/Framework/TestCase.php(940): PHPUnit\Framework\TestCase->runTest()
#6 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/Framework/TestResult.php(698): PHPUnit\Framework\TestCase->runBare()
#7 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/Framework/TestCase.php(895): PHPUnit\Framework\TestResult->run(Object(Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest))
#8 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/Framework/TestSuite.php(755): PHPUnit\Framework\TestCase->run(Object(PHPUnit\Framework\TestResult))
#9 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/Framework/TestSuite.php(755): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
#10 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/Framework/TestSuite.php(755): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
#11 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/TextUI/TestRunner.php(546): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
#12 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/TextUI/Command.php(195): PHPUnit\TextUI\TestRunner->doRun(Object(PHPUnit\Framework\TestSuite), Array, true)
#13 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/TextUI/Command.php(148): PHPUnit\TextUI\Command->run(Array, true)
#14 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/phpunit(17): PHPUnit\TextUI\Command::main()
#15 /Users/andrewtch/Documents/Work/maker-bundle/vendor/symfony/phpunit-bridge/bin/simple-phpunit(251): include('/Users/andrewtc...')
#16 {main}
3) Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest::testCommands with data set "controller_basic" (Symfony\Bundle\MakerBundle\Test\MakerTestDetails Object (...))
file_get_contents(/Users/andrewtch/Documents/Work/maker-bundle/tests/tmp/cache/maker_265e327031/maker_265e327031.json): failed to open stream: No such file or directory

/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:235
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestCase.php:63
/Users/andrewtch/Documents/Work/maker-bundle/tests/Maker/FunctionalTest.php:51

Caused by
Exception: Error running command: "php bin/console make:controller  --no-ansi". Output: "
 Choose a name for your controller class (e.g. GrumpyChefController):
 > 

 [ERROR] The file "src/Controller/FooBarController.php" can't be generated      
         because it already exists.                                             

". Error: "" in /Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestProcess.php:55
Stack trace:
#0 /Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php(172): Symfony\Bundle\MakerBundle\Test\MakerTestProcess->run(false)
#1 /Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestCase.php(31): Symfony\Bundle\MakerBundle\Test\MakerTestEnvironment->runMaker()
#2 /Users/andrewtch/Documents/Work/maker-bundle/tests/Maker/FunctionalTest.php(51): Symfony\Bundle\MakerBundle\Test\MakerTestCase->executeMakerCommand(Object(Symfony\Bundle\MakerBundle\Test\MakerTestDetails))
#3 [internal function]: Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest->testCommands(Object(Symfony\Bundle\MakerBundle\Test\MakerTestDetails))
#4 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/Framework/TestCase.php(1072): ReflectionMethod->invokeArgs(Object(Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest), Array)
#5 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/Framework/TestCase.php(940): PHPUnit\Framework\TestCase->runTest()
#6 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/Framework/TestResult.php(698): PHPUnit\Framework\TestCase->runBare()
#7 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/Framework/TestCase.php(895): PHPUnit\Framework\TestResult->run(Object(Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest))
#8 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/Framework/TestSuite.php(755): PHPUnit\Framework\TestCase->run(Object(PHPUnit\Framework\TestResult))
#9 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/Framework/TestSuite.php(755): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
#10 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/Framework/TestSuite.php(755): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
#11 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/TextUI/TestRunner.php(546): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
#12 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/TextUI/Command.php(195): PHPUnit\TextUI\TestRunner->doRun(Object(PHPUnit\Framework\TestSuite), Array, true)
#13 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/TextUI/Command.php(148): PHPUnit\TextUI\Command->run(Array, true)
#14 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/phpunit(17): PHPUnit\TextUI\Command::main()
#15 /Users/andrewtch/Documents/Work/maker-bundle/vendor/symfony/phpunit-bridge/bin/simple-phpunit(251): include('/Users/andrewtc...')
#16 {main}
4) Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest::testCommands with data set "controller_sub_namespace" (Symfony\Bundle\MakerBundle\Test\MakerTestDetails Object (...))
file_get_contents(/Users/andrewtch/Documents/Work/maker-bundle/tests/tmp/cache/maker_265e327031/maker_265e327031.json): failed to open stream: No such file or directory

/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:235
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestCase.php:63
/Users/andrewtch/Documents/Work/maker-bundle/tests/Maker/FunctionalTest.php:51

Caused by
Exception: Error running command: "php bin/console make:controller  --no-ansi". Output: "
 Choose a name for your controller class (e.g. DeliciousKangarooController):
 > 

 [ERROR] The file "src/Controller/Admin/FooBarController.php" can't be generated
         because it already exists.                                             

". Error: "" in /Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestProcess.php:55
Stack trace:
#0 /Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php(172): Symfony\Bundle\MakerBundle\Test\MakerTestProcess->run(false)
#1 /Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestCase.php(31): Symfony\Bundle\MakerBundle\Test\MakerTestEnvironment->runMaker()
#2 /Users/andrewtch/Documents/Work/maker-bundle/tests/Maker/FunctionalTest.php(51): Symfony\Bundle\MakerBundle\Test\MakerTestCase->executeMakerCommand(Object(Symfony\Bundle\MakerBundle\Test\MakerTestDetails))
#3 [internal function]: Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest->testCommands(Object(Symfony\Bundle\MakerBundle\Test\MakerTestDetails))
#4 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/Framework/TestCase.php(1072): ReflectionMethod->invokeArgs(Object(Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest), Array)
#5 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/Framework/TestCase.php(940): PHPUnit\Framework\TestCase->runTest()
#6 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/Framework/TestResult.php(698): PHPUnit\Framework\TestCase->runBare()
#7 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/Framework/TestCase.php(895): PHPUnit\Framework\TestResult->run(Object(Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest))
#8 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/Framework/TestSuite.php(755): PHPUnit\Framework\TestCase->run(Object(PHPUnit\Framework\TestResult))
#9 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/Framework/TestSuite.php(755): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
#10 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/Framework/TestSuite.php(755): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
#11 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/TextUI/TestRunner.php(546): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
#12 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/TextUI/Command.php(195): PHPUnit\TextUI\TestRunner->doRun(Object(PHPUnit\Framework\TestSuite), Array, true)
#13 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/src/TextUI/Command.php(148): PHPUnit\TextUI\Command->run(Array, true)
#14 /Users/andrewtch/Documents/Work/maker-bundle/vendor/bin/.phpunit/phpunit-6.5/phpunit(17): PHPUnit\TextUI\Command::main()
#15 /Users/andrewtch/Documents/Work/maker-bundle/vendor/symfony/phpunit-bridge/bin/simple-phpunit(251): include('/Users/andrewtc...')
#16 {main}
5) Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest::testEntityCommands with data set "entity_new" (Symfony\Bundle\MakerBundle\Test\MakerTestDetails Object (...))
Exception: Could not find "mysql://db_user:db_password@127.0.0.1:3306/db_name" inside "phpunit.xml.dist"

/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:334
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:116
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestCase.php:27
/Users/andrewtch/Documents/Work/maker-bundle/tests/Maker/FunctionalTest.php:63

6) Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest::testEntityCommands with data set "entity_with_fields" (Symfony\Bundle\MakerBundle\Test\MakerTestDetails Object (...))
Exception: Could not find "mysql://db_user:db_password@127.0.0.1:3306/db_name" inside "phpunit.xml.dist"

/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:334
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:116
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestCase.php:27
/Users/andrewtch/Documents/Work/maker-bundle/tests/Maker/FunctionalTest.php:63

7) Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest::testEntityCommands with data set "entity_updating" (Symfony\Bundle\MakerBundle\Test\MakerTestDetails Object (...))
Exception: Could not find "mysql://db_user:db_password@127.0.0.1:3306/db_name" inside "phpunit.xml.dist"

/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:334
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:116
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestCase.php:27
/Users/andrewtch/Documents/Work/maker-bundle/tests/Maker/FunctionalTest.php:63

8) Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest::testEntityCommands with data set "entity_many_to_one_simple_with_inverse" (Symfony\Bundle\MakerBundle\Test\MakerTestDetails Object (...))
Exception: Could not find "mysql://db_user:db_password@127.0.0.1:3306/db_name" inside "phpunit.xml.dist"

/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:334
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:116
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestCase.php:27
/Users/andrewtch/Documents/Work/maker-bundle/tests/Maker/FunctionalTest.php:63

9) Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest::testEntityCommands with data set "entity_many_to_one_simple_no_inverse" (Symfony\Bundle\MakerBundle\Test\MakerTestDetails Object (...))
Exception: Could not find "mysql://db_user:db_password@127.0.0.1:3306/db_name" inside "phpunit.xml.dist"

/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:334
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:116
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestCase.php:27
/Users/andrewtch/Documents/Work/maker-bundle/tests/Maker/FunctionalTest.php:63

10) Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest::testEntityCommands with data set "entity_one_to_many_simple" (Symfony\Bundle\MakerBundle\Test\MakerTestDetails Object (...))
Exception: Could not find "mysql://db_user:db_password@127.0.0.1:3306/db_name" inside "phpunit.xml.dist"

/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:334
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:116
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestCase.php:27
/Users/andrewtch/Documents/Work/maker-bundle/tests/Maker/FunctionalTest.php:63

11) Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest::testEntityCommands with data set "entity_many_to_many_simple" (Symfony\Bundle\MakerBundle\Test\MakerTestDetails Object (...))
Exception: Could not find "mysql://db_user:db_password@127.0.0.1:3306/db_name" inside "phpunit.xml.dist"

/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:334
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:116
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestCase.php:27
/Users/andrewtch/Documents/Work/maker-bundle/tests/Maker/FunctionalTest.php:63

12) Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest::testEntityCommands with data set "entity_one_to_one_simple" (Symfony\Bundle\MakerBundle\Test\MakerTestDetails Object (...))
Exception: Could not find "mysql://db_user:db_password@127.0.0.1:3306/db_name" inside "phpunit.xml.dist"

/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:334
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:116
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestCase.php:27
/Users/andrewtch/Documents/Work/maker-bundle/tests/Maker/FunctionalTest.php:63

13) Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest::testEntityCommands with data set "entity_many_to_one_vendor_target" (Symfony\Bundle\MakerBundle\Test\MakerTestDetails Object (...))
Exception: Could not find "mysql://db_user:db_password@127.0.0.1:3306/db_name" inside "phpunit.xml.dist"

/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:334
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:116
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestCase.php:27
/Users/andrewtch/Documents/Work/maker-bundle/tests/Maker/FunctionalTest.php:63

14) Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest::testEntityCommands with data set "entity_many_to_many_vendor_target" (Symfony\Bundle\MakerBundle\Test\MakerTestDetails Object (...))
Exception: Could not find "mysql://db_user:db_password@127.0.0.1:3306/db_name" inside "phpunit.xml.dist"

/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:334
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:116
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestCase.php:27
/Users/andrewtch/Documents/Work/maker-bundle/tests/Maker/FunctionalTest.php:63

15) Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest::testEntityCommands with data set "entity_one_to_one_vendor_target" (Symfony\Bundle\MakerBundle\Test\MakerTestDetails Object (...))
Exception: Could not find "mysql://db_user:db_password@127.0.0.1:3306/db_name" inside "phpunit.xml.dist"

/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:334
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:116
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestCase.php:27
/Users/andrewtch/Documents/Work/maker-bundle/tests/Maker/FunctionalTest.php:63

16) Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest::testEntityCommands with data set "entity_regenerate" (Symfony\Bundle\MakerBundle\Test\MakerTestDetails Object (...))
Exception: Could not find "mysql://db_user:db_password@127.0.0.1:3306/db_name" inside "phpunit.xml.dist"

/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:334
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:116
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestCase.php:27
/Users/andrewtch/Documents/Work/maker-bundle/tests/Maker/FunctionalTest.php:63

17) Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest::testEntityCommands with data set "entity_regenerate_embeddable" (Symfony\Bundle\MakerBundle\Test\MakerTestDetails Object (...))
Exception: Could not find "mysql://db_user:db_password@127.0.0.1:3306/db_name" inside "phpunit.xml.dist"

/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:334
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:116
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestCase.php:27
/Users/andrewtch/Documents/Work/maker-bundle/tests/Maker/FunctionalTest.php:63

18) Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest::testEntityCommands with data set "entity_regenerate_overwrite" (Symfony\Bundle\MakerBundle\Test\MakerTestDetails Object (...))
Exception: Could not find "mysql://db_user:db_password@127.0.0.1:3306/db_name" inside "phpunit.xml.dist"

/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:334
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:116
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestCase.php:27
/Users/andrewtch/Documents/Work/maker-bundle/tests/Maker/FunctionalTest.php:63

19) Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest::testEntityCommands with data set "entity_regenerate_xml" (Symfony\Bundle\MakerBundle\Test\MakerTestDetails Object (...))
Exception: Could not find "mysql://db_user:db_password@127.0.0.1:3306/db_name" inside "phpunit.xml.dist"

/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:334
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:116
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestCase.php:27
/Users/andrewtch/Documents/Work/maker-bundle/tests/Maker/FunctionalTest.php:63

20) Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest::testEntityCommands with data set "entity_xml_mapping_error_existing" (Symfony\Bundle\MakerBundle\Test\MakerTestDetails Object (...))
Exception: Could not find "type: annotation" inside "config/packages/doctrine.yaml"

/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:334
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:116
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestCase.php:27
/Users/andrewtch/Documents/Work/maker-bundle/tests/Maker/FunctionalTest.php:63

21) Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest::testEntityCommands with data set "entity_xml_mapping_error_new_class" (Symfony\Bundle\MakerBundle\Test\MakerTestDetails Object (...))
Exception: Could not find "type: annotation" inside "config/packages/doctrine.yaml"

/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:334
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:116
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestCase.php:27
/Users/andrewtch/Documents/Work/maker-bundle/tests/Maker/FunctionalTest.php:63

22) Symfony\Bundle\MakerBundle\Tests\Maker\FunctionalTest::testEntityCommands with data set "entity_updating_overwrite" (Symfony\Bundle\MakerBundle\Test\MakerTestDetails Object (...))
Exception: Error running command: "php bin/console make:entity --overwrite --no-ansi". Output: "
 Class name of the entity to create or update (e.g. BravePizza):
 > 

 [ERROR] Only annotation mapping is supported by make:entity, but the           
         <info>App\Entity\User</info> class uses a different format. If you     
         would like this command to generate the properties & getter/setter     
         methods, add your mapping configuration, and then re-run this command  
         with the <info>--regenerate</info> flag.                               

". Error: ""

/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestProcess.php:55
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestEnvironment.php:172
/Users/andrewtch/Documents/Work/maker-bundle/src/Test/MakerTestCase.php:31
/Users/andrewtch/Documents/Work/maker-bundle/tests/Maker/FunctionalTest.php:63

ERRORS!
Tests: 175, Assertions: 315, Errors: 22.
```

I'm using 7.2.4. on Mac, the same commit as here - the same 5110979fa610202242db6c3acd0f2804aeb3e349 commit as here: https://travis-ci.org/symfony/maker-bundle/builds/375426623 which runs just fine.

What am I missing here?

--- a/src/Doctrine/EntityRegenerator.php
+++ b/src/Doctrine/EntityRegenerator.php
@@ -66,6 +66,8 @@ final class EntityRegenerator
                 $classPath = $this->getPathOfClass($classMetadata->name);
             }
 
+            $mappedFields = $this->getMappedFieldsInEntity($classMetadata);
+
             if ($classMetadata->customRepositoryClassName) {
                 $this->generateRepository($classMetadata);
             }
@@ -95,6 +97,10 @@ final class EntityRegenerator
                     continue;
                 }
 
+                if (!in_array($fieldName, $mappedFields)) {
+                    continue;
+                }
+
                 $manipulator->addEntityField($fieldName, $mapping);
             }
 
@@ -108,6 +114,10 @@ final class EntityRegenerator
             };
 
             foreach ($classMetadata->associationMappings as $fieldName => $mapping) {
+                if (!in_array($fieldName, $mappedFields)) {
+                    continue;
+                }
+
                 switch ($mapping['type']) {
                     case ClassMetadata::MANY_TO_ONE:
                         $relation = (new RelationManyToOne())
@@ -226,5 +236,37 @@ final class EntityRegenerator
         );
 
         $this->generator->writeChanges();
+    }
+
+    private function getMappedFieldsInEntity(ClassMetadata $classMetadata)
+    {
+        /* @var $classReflection \ReflectionClass */
+        $classReflection = $classMetadata->reflClass;
+
+        $targetFields = array_merge(
+            array_keys($classMetadata->fieldMappings),
+            array_keys($classMetadata->associationMappings)
+        );
+
+        if ($classReflection) {
+            // exclude traits
+            $traitProperties = [];
+
+            foreach ($classReflection->getTraits() as $trait) {
+                foreach ($trait->getProperties() as $property) {
+                    $traitProperties[] = $property->getName();
+                }
+            }
+
+            $targetFields = array_diff($targetFields, $traitProperties);
+
+            // exclude inherited properties
+            $targetFields = array_filter($targetFields, function ($field) use ($classReflection) {
+                return $classReflection->hasProperty($field) &&
+                    $classReflection->getProperty($field)->getDeclaringClass()->getName() == $classReflection->getName();
+            });
+        }
+
+        return $targetFields;
     }
 }

--- a/src/Util/ClassSourceManipulator.php
+++ b/src/Util/ClassSourceManipulator.php
@@ -535,7 +535,20 @@ final class ClassSourceManipulator
     private function addStatementToConstructor(Node\Stmt $stmt)
     {
         if (!$this->getConstructorNode()) {
-            $constructorNode = $getterNodeBuilder = (new Builder\Method('__construct'))->makePublic()->getNode();
+            $constructorNode = (new Builder\Method('__construct'))->makePublic()->getNode();
+
+            // add call to parent::__construct() if there is a need to
+            try {
+                $ref = new \ReflectionClass($this->getThisFullClassName());
+
+                if ($ref->getParentClass() && $ref->getParentClass()->getConstructor()) {
+                    $constructorNode->stmts[] = new Node\Stmt\Expression(
+                        new Node\Expr\StaticCall(new Node\Name('parent'), new Node\Identifier('__construct'))
+                    );
+                }
+            } catch (\ReflectionException $e) {
+            }
+
             $this->addNodeAfterProperties($constructorNode);
         }
 

--- a/tests/Doctrine/EntityRegeneratorTest.php
+++ b/tests/Doctrine/EntityRegeneratorTest.php
@@ -69,7 +69,9 @@ class EntityRegeneratorTest extends TestCase
         $fs = new Filesystem();
         $tmpDir = __DIR__.'/../tmp/current_project';
         $fs->remove($tmpDir);
-        $fs->mirror($sourceDir, $tmpDir);
+
+        // if traits (Timestampable, Teamable) gets copied into new project, tests will fail because of double exclusion
+        $fs->mirror($sourceDir, $tmpDir, $this->createAllButTraitsIterator($sourceDir));
 
         $kernel->boot();
         $container = $kernel->getContainer();
@@ -113,6 +115,13 @@ class EntityRegeneratorTest extends TestCase
 
             $this->assertEquals($expectedContents, $actualContents, sprintf('File "%s" does not match: %s', $file->getFilename(), $actualContents));
         }
+    }
+
+    private function createAllButTraitsIterator(string $sourceDir): \Iterator
+    {
+        $directoryIterator = new \RecursiveDirectoryIterator($sourceDir, \FilesystemIterator::SKIP_DOTS);
+        $filter = new AllButTraitsIterator($directoryIterator);
+        return new \RecursiveIteratorIterator($filter, \RecursiveIteratorIterator::SELF_FIRST);
     }
 }
 
@@ -202,5 +211,12 @@ class TestXmlEntityRegeneratorKernel extends Kernel
     public function getRootDir()
     {
         return __DIR__.'/../tmp/current_project';
+    }
+}
+
+class AllButTraitsIterator extends \RecursiveFilterIterator
+{
+    public function accept() {
+        return !in_array($this->current()->getFilename(), ['TeamTrait.php', 'TimestampableTrait.php']);
     }
 }

--- a/tests/Doctrine/fixtures/expected_no_overwrite/src/Entity/BaseClient.php
+++ b/tests/Doctrine/fixtures/expected_no_overwrite/src/Entity/BaseClient.php
@@ -1,0 +1,81 @@
+<?php
+
+namespace Symfony\Bundle\MakerBundle\Tests\Doctrine\fixtures\source_project\src\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\MappedSuperclass()
+ */
+class BaseClient
+{
+    use TeamTrait;
+
+    /**
+     * @ORM\Id
+     * @ORM\GeneratedValue
+     * @ORM\Column(type="integer")
+     */
+    private $id;
+
+    /**
+     * @ORM\Column(type="string")
+     */
+    private $name;
+
+    /**
+     * @ORM\ManyToOne(targetEntity="User")
+     */
+    private $creator;
+
+    /**
+     * @ORM\Column(type="integer")
+     */
+    private $magic;
+
+    public function __construct()
+    {
+        $this->magic = 42;
+    }
+
+    public function getId()
+    {
+        return $this->id;
+    }
+
+    public function getName(): ?string
+    {
+        return $this->name;
+    }
+
+    public function setName(string $name): self
+    {
+        $this->name = $name;
+
+        return $this;
+    }
+
+    public function getMagic(): ?int
+    {
+        return $this->magic;
+    }
+
+    public function setMagic(int $magic): self
+    {
+        $this->magic = $magic;
+
+        return $this;
+    }
+
+    public function getCreator(): ?User
+    {
+        return $this->creator;
+    }
+
+    public function setCreator(?User $creator): self
+    {
+        $this->creator = $creator;
+
+        return $this;
+    }
+}

--- a/tests/Doctrine/fixtures/expected_no_overwrite/src/Entity/Client.php
+++ b/tests/Doctrine/fixtures/expected_no_overwrite/src/Entity/Client.php
@@ -1,0 +1,88 @@
+<?php
+
+namespace Symfony\Bundle\MakerBundle\Tests\Doctrine\fixtures\source_project\src\Entity;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity()
+ */
+class Client extends BaseClient
+{
+    use TimestampableTrait;
+
+    /**
+     * @ORM\Column(type="string")
+     * @var string
+     */
+    private $apiKey;
+
+    /**
+     * @ORM\ManyToMany(targetEntity="Tag")
+     */
+    private $tags;
+
+    /**
+     * @ORM\Embedded(class="Embed")
+     */
+    private $embed;
+
+    public function __construct()
+    {
+        parent::__construct();
+        $this->embed = new Embed();
+        $this->tags = new ArrayCollection();
+    }
+
+    public function getEmbed(): Embed
+    {
+        return $this->embed;
+    }
+
+    public function setEmbed(Embed $embed): self
+    {
+        $this->embed = $embed;
+
+        return $this;
+    }
+
+    public function getApiKey(): ?string
+    {
+        return $this->apiKey;
+    }
+
+    public function setApiKey(string $apiKey): self
+    {
+        $this->apiKey = $apiKey;
+
+        return $this;
+    }
+
+    /**
+     * @return Collection|Tag[]
+     */
+    public function getTags(): Collection
+    {
+        return $this->tags;
+    }
+
+    public function addTag(Tag $tag): self
+    {
+        if (!$this->tags->contains($tag)) {
+            $this->tags[] = $tag;
+        }
+
+        return $this;
+    }
+
+    public function removeTag(Tag $tag): self
+    {
+        if ($this->tags->contains($tag)) {
+            $this->tags->removeElement($tag);
+        }
+
+        return $this;
+    }
+}

--- a/tests/Doctrine/fixtures/expected_no_overwrite/src/Entity/Embed.php
+++ b/tests/Doctrine/fixtures/expected_no_overwrite/src/Entity/Embed.php
@@ -1,0 +1,28 @@
+<?php
+
+namespace Symfony\Bundle\MakerBundle\Tests\Doctrine\fixtures\source_project\src\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Embeddable()
+ */
+class Embed
+{
+    /**
+     * @ORM\Column(type="integer")
+     */
+    private $val;
+
+    public function getVal(): ?int
+    {
+        return $this->val;
+    }
+
+    public function setVal(int $val): self
+    {
+        $this->val = $val;
+
+        return $this;
+    }
+}

--- a/tests/Doctrine/fixtures/expected_overwrite/src/Entity/BaseClient.php
+++ b/tests/Doctrine/fixtures/expected_overwrite/src/Entity/BaseClient.php
@@ -1,0 +1,81 @@
+<?php
+
+namespace Symfony\Bundle\MakerBundle\Tests\Doctrine\fixtures\source_project\src\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\MappedSuperclass()
+ */
+class BaseClient
+{
+    use TeamTrait;
+
+    /**
+     * @ORM\Id
+     * @ORM\GeneratedValue
+     * @ORM\Column(type="integer")
+     */
+    private $id;
+
+    /**
+     * @ORM\Column(type="string")
+     */
+    private $name;
+
+    /**
+     * @ORM\ManyToOne(targetEntity="User")
+     */
+    private $creator;
+
+    /**
+     * @ORM\Column(type="integer")
+     */
+    private $magic;
+
+    public function __construct()
+    {
+        $this->magic = 42;
+    }
+
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    public function getName(): ?string
+    {
+        return $this->name;
+    }
+
+    public function setName(string $name): self
+    {
+        $this->name = $name;
+
+        return $this;
+    }
+
+    public function getMagic(): ?int
+    {
+        return $this->magic;
+    }
+
+    public function setMagic(int $magic): self
+    {
+        $this->magic = $magic;
+
+        return $this;
+    }
+
+    public function getCreator(): ?User
+    {
+        return $this->creator;
+    }
+
+    public function setCreator(?User $creator): self
+    {
+        $this->creator = $creator;
+
+        return $this;
+    }
+}

--- a/tests/Doctrine/fixtures/expected_overwrite/src/Entity/Client.php
+++ b/tests/Doctrine/fixtures/expected_overwrite/src/Entity/Client.php
@@ -1,0 +1,88 @@
+<?php
+
+namespace Symfony\Bundle\MakerBundle\Tests\Doctrine\fixtures\source_project\src\Entity;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity()
+ */
+class Client extends BaseClient
+{
+    use TimestampableTrait;
+
+    /**
+     * @ORM\Column(type="string")
+     * @var string
+     */
+    private $apiKey;
+
+    /**
+     * @ORM\ManyToMany(targetEntity="Tag")
+     */
+    private $tags;
+
+    /**
+     * @ORM\Embedded(class="Embed")
+     */
+    private $embed;
+
+    public function __construct()
+    {
+        parent::__construct();
+        $this->embed = new Embed();
+        $this->tags = new ArrayCollection();
+    }
+
+    public function getEmbed(): Embed
+    {
+        return $this->embed;
+    }
+
+    public function setEmbed(Embed $embed): self
+    {
+        $this->embed = $embed;
+
+        return $this;
+    }
+
+    public function getApiKey(): ?string
+    {
+        return $this->apiKey;
+    }
+
+    public function setApiKey(string $apiKey): self
+    {
+        $this->apiKey = $apiKey;
+
+        return $this;
+    }
+
+    /**
+     * @return Collection|Tag[]
+     */
+    public function getTags(): Collection
+    {
+        return $this->tags;
+    }
+
+    public function addTag(Tag $tag): self
+    {
+        if (!$this->tags->contains($tag)) {
+            $this->tags[] = $tag;
+        }
+
+        return $this;
+    }
+
+    public function removeTag(Tag $tag): self
+    {
+        if ($this->tags->contains($tag)) {
+            $this->tags->removeElement($tag);
+        }
+
+        return $this;
+    }
+}

--- a/tests/Doctrine/fixtures/expected_overwrite/src/Entity/Embed.php
+++ b/tests/Doctrine/fixtures/expected_overwrite/src/Entity/Embed.php
@@ -1,0 +1,28 @@
+<?php
+
+namespace Symfony\Bundle\MakerBundle\Tests\Doctrine\fixtures\source_project\src\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Embeddable()
+ */
+class Embed
+{
+    /**
+     * @ORM\Column(type="integer")
+     */
+    private $val;
+
+    public function getVal(): ?int
+    {
+        return $this->val;
+    }
+
+    public function setVal(int $val): self
+    {
+        $this->val = $val;
+
+        return $this;
+    }
+}

--- a/tests/Doctrine/fixtures/source_project/src/Entity/BaseClient.php
+++ b/tests/Doctrine/fixtures/source_project/src/Entity/BaseClient.php
@@ -1,0 +1,45 @@
+<?php
+
+namespace Symfony\Bundle\MakerBundle\Tests\Doctrine\fixtures\source_project\src\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\MappedSuperclass()
+ */
+class BaseClient
+{
+    use TeamTrait;
+
+    /**
+     * @ORM\Id
+     * @ORM\GeneratedValue
+     * @ORM\Column(type="integer")
+     */
+    private $id;
+
+    /**
+     * @ORM\Column(type="string")
+     */
+    private $name;
+
+    /**
+     * @ORM\ManyToOne(targetEntity="User")
+     */
+    private $creator;
+
+    /**
+     * @ORM\Column(type="integer")
+     */
+    private $magic;
+
+    public function __construct()
+    {
+        $this->magic = 42;
+    }
+
+    public function getId()
+    {
+        return $this->id;
+    }
+}

--- a/tests/Doctrine/fixtures/source_project/src/Entity/Client.php
+++ b/tests/Doctrine/fixtures/source_project/src/Entity/Client.php
@@ -1,0 +1,29 @@
+<?php
+
+namespace Symfony\Bundle\MakerBundle\Tests\Doctrine\fixtures\source_project\src\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity()
+ */
+class Client extends BaseClient
+{
+    use TimestampableTrait;
+
+    /**
+     * @ORM\Column(type="string")
+     * @var string
+     */
+    private $apiKey;
+
+    /**
+     * @ORM\ManyToMany(targetEntity="Tag")
+     */
+    private $tags;
+
+    /**
+     * @ORM\Embedded(class="Embed")
+     */
+    private $embed;
+}

--- a/tests/Doctrine/fixtures/source_project/src/Entity/Embed.php
+++ b/tests/Doctrine/fixtures/source_project/src/Entity/Embed.php
@@ -1,0 +1,16 @@
+<?php
+
+namespace Symfony\Bundle\MakerBundle\Tests\Doctrine\fixtures\source_project\src\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Embeddable()
+ */
+class Embed
+{
+    /**
+     * @ORM\Column(type="integer")
+     */
+    private $val;
+}

--- a/tests/Doctrine/fixtures/source_project/src/Entity/TeamTrait.php
+++ b/tests/Doctrine/fixtures/source_project/src/Entity/TeamTrait.php
@@ -1,0 +1,13 @@
+<?php
+
+namespace Symfony\Bundle\MakerBundle\Tests\Doctrine\fixtures\source_project\src\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+trait TeamTrait
+{
+    /**
+     * @ORM\ManyToMany(targetEntity="User")
+     */
+    private $members;
+}

--- a/tests/Doctrine/fixtures/source_project/src/Entity/TimestampableTrait.php
+++ b/tests/Doctrine/fixtures/source_project/src/Entity/TimestampableTrait.php
@@ -1,0 +1,50 @@
+<?php
+
+namespace Symfony\Bundle\MakerBundle\Tests\Doctrine\fixtures\source_project\src\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+trait TimestampableTrait
+{
+    /**
+     * @ORM\Column(type="datetime")
+     */
+    private $createdAt;
+
+    /**
+     * @ORM\Column(type="datetime")
+     */
+    private $updatedAt;
+
+    /**
+     * @return mixed
+     */
+    public function getCreatedAt()
+    {
+        return $this->createdAt;
+    }
+
+    /**
+     * @param mixed $createdAt
+     */
+    public function setCreatedAt($createdAt): void
+    {
+        $this->createdAt = $createdAt;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getUpdatedAt()
+    {
+        return $this->updatedAt;
+    }
+
+    /**
+     * @param mixed $updatedAt
+     */
+    public function setUpdatedAt($updatedAt): void
+    {
+        $this->updatedAt = $updatedAt;
+    }
+}

